File: recovered-source/jadx/sources/me/ag2s/epublib/domain/EpubBook.java
package me.ag2s.epublib.domain;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/* JADX INFO: loaded from: reader-pro-classes-3.2.14.jar:me/ag2s/epublib/domain/EpubBook.class */
public class EpubBook implements Serializable {
    private static final long serialVersionUID = 2068355170895770100L;
    private Resource opfResource;
    private Resource ncxResource;
    private Resource coverImage;
    private Resources resources = new Resources();
    private Metadata metadata = new Metadata();
    private Spine spine = new Spine();
    private TableOfContents tableOfContents = new TableOfContents();
    private final Guide guide = new Guide();
    private String version = "2.0";

    public String getVersion() {
        return this.version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public boolean isEpub3() {
        return this.version.startsWith("3.");
    }

    public TOCReference addSection(TOCReference parentSection, String sectionTitle, Resource resource) {
        return addSection(parentSection, sectionTitle, resource, null);
    }

    public TOCReference addSection(TOCReference parentSection, String sectionTitle, Resource resource, String fragmentId) {
        getResources().add(resource);
        if (this.spine.findFirstResourceById(resource.getId()) < 0) {
            this.spine.addSpineReference(new SpineReference(resource));
        }
        return parentSection.addChildSection(new TOCReference(sectionTitle, resource, fragmentId));
    }

    public TOCReference addSection(String title, Resource resource) {
        return addSection(title, resource, (String) null);
    }

    public TOCReference addSection(String title, Resource resource, String fragmentId) {
        getResources().add(resource);
        TOCReference tocReference = this.tableOfContents.addTOCReference(new TOCReference(title, resource, fragmentId));
        if (this.spine.findFirstResourceById(resource.getId()) < 0) {
            this.spine.addSpineReference(new SpineReference(resource));
        }
        return tocReference;
    }

    public void generateSpineFromTableOfContents() {
        Spine spine = new Spine(this.tableOfContents);
        spine.setTocResource(this.spine.getTocResource());
        this.spine = spine;
    }

    public Metadata getMetadata() {
        return this.metadata;
    }

    public void setMetadata(Metadata metadata) {
        this.metadata = metadata;
    }

    public void setResources(Resources resources) {
        this.resources = resources;
    }

    public Resource addResource(Resource resource) {
        return this.resources.add(resource);
    }

    public Resources getResources() {
        return this.resources;
    }

    public Spine getSpine() {
        return this.spine;
    }

    public void setSpine(Spine spine) {
        this.spine = spine;
    }

    public TableOfContents getTableOfContents() {
        return this.tableOfContents;
    }

    public void setTableOfContents(TableOfContents tableOfContents) {
        this.tableOfContents = tableOfContents;
    }

    public Resource getCoverPage() {
        Resource coverPage = this.guide.getCoverPage();
        if (coverPage == null) {
            coverPage = this.spine.getResource(0);
        }
        return coverPage;
    }

    public void setCoverPage(Resource coverPage) {
        if (coverPage == null) {
            return;
        }
        if (this.resources.notContainsByHref(coverPage.getHref())) {
            this.resources.add(coverPage);
        }
        this.guide.setCoverPage(coverPage);
    }

    public String getTitle() {
        return getMetadata().getFirstTitle();
    }

    public Resource getCoverImage() {
        return this.coverImage;
    }

    public void setCoverImage(Resource coverImage) {
        if (coverImage == null) {
            return;
        }
        if (this.resources.notContainsByHref(coverImage.getHref())) {
            this.resources.add(coverImage);
        }
        this.coverImage = coverImage;
    }

    public Guide getGuide() {
        return this.guide;
    }

    public List<Resource> getContents() {
        Map<String, Resource> result = new LinkedHashMap<>();
        addToContentsResult(getCoverPage(), result);
        for (SpineReference spineReference : getSpine().getSpineReferences()) {
            addToContentsResult(spineReference.getResource(), result);
        }
        for (Resource resource : getTableOfContents().getAllUniqueResources()) {
            addToContentsResult(resource, result);
        }
        for (GuideReference guideReference : getGuide().getReferences()) {
            addToContentsResult(guideReference.getResource(), result);
        }
        return new ArrayList(result.values());
    }

    private static void addToContentsResult(Resource resource, Map<String, Resource> allReachableResources) {
        if (resource != null && !allReachableResources.containsKey(resource.getHref())) {
            allReachableResources.put(resource.getHref(), resource);
        }
    }

    public Resource getOpfResource() {
        return this.opfResource;
    }

    public void setOpfResource(Resource opfResource) {
        this.opfResource = opfResource;
    }

    public void setNcxResource(Resource ncxResource) {
        this.ncxResource = ncxResource;
    }

    public Resource getNcxResource() {
        return this.ncxResource;
    }
}

File: src/main/kotlin/me/ag2s/epublib/domain/EpubBook.kt
package me.ag2s.epublib.domain

import java.io.Serializable

class EpubBook : Serializable {
    private var resources = Resources(); private var metadata = Metadata(); private var spine = Spine(); private var tableOfContents = TableOfContents(); private val guide = Guide(); private var opfResource: Resource? = null; private var ncxResource: Resource? = null; private var coverImage: Resource? = null; private var version = "2.0"
    fun getVersion() = version; fun setVersion(value: String) { version = value }; fun isEpub3() = version.startsWith("3.")
    fun addSection(parent: TOCReference, title: String, resource: Resource) = addSection(parent, title, resource, null)
    fun addSection(parent: TOCReference, title: String, resource: Resource, fragmentId: String?): TOCReference { resources.add(resource); if (spine.findFirstResourceById(resource.getId()) < 0) spine.addSpineReference(SpineReference(resource)); return parent.addChildSection(TOCReference(title, resource, fragmentId)) }
    fun addSection(title: String, resource: Resource) = addSection(title, resource, null)
    fun addSection(title: String, resource: Resource, fragmentId: String?): TOCReference { resources.add(resource); val reference = tableOfContents.addTOCReference(TOCReference(title, resource, fragmentId)); if (spine.findFirstResourceById(resource.getId()) < 0) spine.addSpineReference(SpineReference(resource)); return reference }
    fun generateSpineFromTableOfContents() { val generated = Spine(tableOfContents); generated.setTocResource(spine.getTocResource()); spine = generated }
    fun getMetadata() = metadata; fun setMetadata(value: Metadata) { metadata = value }; fun setResources(value: Resources) { resources = value }; fun addResource(value: Resource) = resources.add(value); fun getResources() = resources; fun getSpine() = spine; fun setSpine(value: Spine) { spine = value }; fun getTableOfContents() = tableOfContents; fun setTableOfContents(value: TableOfContents) { tableOfContents = value }
    fun getCoverPage(): Resource? = guide.getCoverPage() ?: spine.getResource(0)
    fun setCoverPage(value: Resource?) { if (value != null) { if (resources.notContainsByHref(value.getHref())) resources.add(value); guide.setCoverPage(value) } }
    fun getTitle() = metadata.getFirstTitle(); fun getCoverImage() = coverImage; fun setCoverImage(value: Resource?) { if (value != null) { if (resources.notContainsByHref(value.getHref())) resources.add(value); coverImage = value } }; fun getGuide() = guide
    fun getContents(): MutableList<Resource> { val result = LinkedHashMap<String?, Resource>(); fun add(value: Resource?) { if (value != null && !result.containsKey(value.getHref())) result[value.getHref()] = value }; add(getCoverPage()); spine.getSpineReferences().forEach { add(it.getResource()) }; tableOfContents.getAllUniqueResources().forEach { add(it) }; guide.getReferences().forEach { add(it.getResource()) }; return ArrayList(result.values) }
    fun getOpfResource() = opfResource; fun setOpfResource(value: Resource?) { opfResource = value }; fun setNcxResource(value: Resource?) { ncxResource = value }; fun getNcxResource() = ncxResource
    companion object { private const val serialVersionUID = 2068355170895770100L }
}
